src/main/groovy/graph/Graph.groovy

```diff
@@ -131,23 +131,29 @@ class Graph {
      * @param closure
      * @return the resulting vertex
      */
-    def vertex(map, Closure closure = null) {
-        def vertex = vertices[map.name] ?: vertexFactory.newVertex(map.name)
-
-        vertex = map.traits?.inject(vertex) { val, it ->
-            val.delegateAs(it)
-        } ?: vertex
+    Vertex vertex(Map<String, ?> map, @DelegatesTo(VertexSpec) Closure closure = null) {
+        Vertex vertex = vertices[map.name] ?: vertexFactory.newVertex(map.name)
+        VertexSpec spec = new VertexSpec(name:vertex.name, vertex:vertex)
+        if (map.traits) {
+            spec.traits(map.traits as Class[])
+        }
+        if (map.connectsTo) {
+            spec.connectsTo(map.connectsTo as String[])
+        }
 
-        if (map.trait) {
-            vertex.delegateAs(map.trait)
+        if (closure) {
+            Closure code = closure.rehydrate(spec, this, this)
+            code()
         }
 
-        if (closure != null) {
-            closure.delegate = vertex
-            closure()
+        if (spec.traits) {
+            vertex.delegateAs(spec.traits as Class[])
+        }
+        spec.connectsTo.each {
+            edge vertex.name, it
         }
 
```

Comment:
There needs to be a way to configure traits once the traits are added to the vertex. Using a second closure could work. It could be cool to have something like gradle's doLast with a leftShirt operator.